geometry/region2d.hpp

```diff
@@ -206,7 +218,7 @@ namespace m2
           BigCoordT const delta = prev.y - curr.y;
           BigCoordT const cp = CrossProduct(curr, prev);
 
-          if (!equalF.EqualZero(cp, delta))
+          if (!equalF.EqualZeroSquarePrecision(cp))
```

Comment:
Комментарий в стиле:
```
Squared precision is needed here because of comparison between cross product of two vectors and zero. It's impossible to compare them relatively, so they're compared absolutely, and, as cross product is proportional to product of lengths of both operands precision must be squared too.
```